cupyx/scipy/signal/signaltools.py

```diff
@@ -47,6 +156,109 @@ def _fftconv_faster(x, h, mode):
+
+    .. seealso:: :func:`cupyx.scipy.signal.convolve`
+    .. seealso:: :func:`cupyx.scipy.signal.fftconvolve`
+    .. seealso:: :func:`cupyx.scipy.signal.oaconvolve`
+    .. seealso:: :func:`cupyx.scipy.signal.correlate2d`
+    .. seealso:: :func:`cupyx.scipy.ndimage.convolve`
+    .. seealso:: :func:`scipy.signal.convolve2d`
+    """
+    return _correlate2d(in1, in2, mode, boundary, fillvalue, True)
+
+
+def correlate2d(in1, in2, mode='full', boundary='fill', fillvalue=0):
+    """Cross-correlate two 2-dimensional arrays.
+
+    Cross correlate ``in1`` and ``in2`` with output size determined by
+    ``mode``, and boundary conditions determined by ``boundary`` and
+    ``fillvalue``.
+
+    Args:
+        in1 (cupy.ndarray): First input.
+        in2 (cupy.ndarray): Second input. Should have the same number of
+            dimensions as ``in1``.
+        mode (str): Indicates the size of the output:
+            ``'full'``: output is the full discrete linear convolution
+                        (default)
+            ``'valid'``: output consists only of those elements that do not
+                         rely on the zero-padding. Either ``in1`` or ``in2``
+                         must be at least as large as the other in every
+                         dimension.
+            ``'same'``: output is the same size as ``in1``, centered
+                      with respect to the 'full' output
+        boundary (str): Indicates how to handle boundaries:
+            ``fill``: pad input arrays with fillvalue (default)
+            ``wrap``: circular boundary conditions
+            ``symm``: symmetrical boundary conditions
+        fillvalue (scalar): Value to fill pad input arrays with. Default is 0.
+
+    Returns:
+        cupy.ndarray: A 2-dimensional array containing a subset of the discrete
+            linear cross-correlation of ``in1`` with ``in2``.
+
+    Note:
+        When using ``"same"`` mode with even-length inputs, the outputs of
+        ``correlate`` and ``correlate2d`` differ: There is a 1-index offset
+        between them.
+
+    .. seealso:: :func:`cupyx.scipy.signal.correlate`
+    .. seealso:: :func:`cupyx.scipy.signal.convolve2d`
+    .. seealso:: :func:`cupyx.scipy.ndimage.correlate`
+    .. seealso:: :func:`scipy.signal.correlate2d`
+    """
+    return _correlate2d(in1, in2, mode, boundary, fillvalue, False)
+
+
+def _correlate2d(in1, in2, mode, boundary, fillvalue, convolution=False):
+    if not (in1.ndim == in2.ndim == 2):
+        raise ValueError('{} inputs must both be 2-D arrays'.format(
+            'convolve2d' if convolution else 'correlate2d'))
+    _boundaries = {
+        'fill': 'constant', 'pad': 'constant',
+        'wrap': 'wrap', 'circular': 'wrap',
+        'symm': 'reflect', 'symmetric': 'reflect',
+    }
+    boundary = _boundaries.get(boundary)
+    if boundary is None:
+        raise ValueError("Acceptable boundary flags are 'fill' (or 'pad'), "
```

Comment:
Why not `fill`, `wrap` and `symm` only here?

Reply:
This mimics the scipy error message which includes the un-documented acceptable boundary flags.